include/spi.h

```diff
@@ -152,25 +183,32 @@ struct spi_cs_control {
  *     mode                [ 1 : 3 ]   - Polarity, phase and loop mode.
  *     transfer            [ 4 ]       - LSB or MSB first.
  *     word_size           [ 5 : 10 ]  - Size of a data frame in bits.
- *     lines               [ 11 : 12 ] - MISO lines: Single/Dual/Quad.
+ *     lines               [ 11 : 12 ] - MISO lines: Single/Dual/Quad/Octal.
  *     cs_hold             [ 13 ]      - Hold on the CS line if possible.
  *     lock_on             [ 14 ]      - Keep resource locked for the caller.
- *     eeprom              [ 15 ]      - EEPROM mode.
+ *     cs_active_high      [ 15 ]      - Active high CS logice.
  * @param slave is the slave number from 0 to host controller slave limit.
  * @param cs is a valid pointer on a struct spi_cs_control is CS line is
  *    emulated through a gpio line, or NULL otherwise.
  *
- * @note cs_hold, lock_on and eeprom_rx can be changed between consecutive
+ * @note cs_hold and lock_on can be changed between consecutive
  * transceive call.
```

Comment:
This note is a bit confusing. Does it imply that other fields of `operation` cannot be changed in consecutive calls?

Reply:
Yes, and it should be specifically said here for clarity.

Reply:
But why? Why couldn't a controller perform one transfer in e.g. mode 0 with word size 8, then another one (perhaps with a different slave but not necessarily) in mode 3 with word size 16?

Reply:
It totally could. But then it means you are not accessing the same slave, is it's another spi_config you need for this.

I mean, I don't see any real use-case where you would be able to switch word-size when dealing with one specific slave. Does that exist?

Reply:
I saw such behavior when a CPU booted from an external flash - first it was trying to read an initial part of the boot image with low frequency and 8 bit word size. When this was successful, it switched to higher frequency and 16-bit words (and changed SPI mode from 0 to 3 as well).
And I can see this as a use case. If you don't know exactly what slave (with what capabilities) you can expect on the bus, first you do a transaction with some "safe" configuration just to collect required info, then you switch settings to the ones providing the best performance.

Reply:
This is a really particular use case, which could be addressed by having 2 spi_config: safe one vs performance one. I am not sure we want to generalize this. Because going through the controller configuration at every transaction means quite a lot of cpu cycles: reading/writing registers etc... Of course we could make controller copying the config directly, and it would do a memcmp, but that's also a little bit under-performing and takes a bit of memory.

Imo, for thus case you describe, 2 spi_config would do.